app/controllers/application_controller/ci_processing.rb

```diff
@@ -1868,6 +1868,7 @@ def process_cloud_object_storage_buttons(pressed)
     task = pressed.sub("#{klass.name.underscore.to_sym}_", "")
 
     return tag(klass) if task == "tag"
+    return javascript_redirect(:controller => klass.name.underscore.to_sym, :action => "new") if task == "new"
 
```

Comment:
@miha-plesko I see that you have already added `delete` code in `ci_processing` and now you are about to add `new`. 
The `delete` handling should have been added in `cloud_object_store_container_controller.rb`

The button handling for `new`, `edit`, `delete` should all live in the acting controller code.
Let me give you another code example that was recently added --
https://github.com/ManageIQ/manageiq-ui-classic/blob/master/app/controllers/ansible_credential_controller.rb#L27-L35
It's clear, concise and easy to follow. Could we implement something like that?